tools/upgrade-to-1.1.0.sh

```diff
@@ -0,0 +1,212 @@
+# installing the new version. This script helps you to do the cleanup. The script expects 
+# the current namespace where the operator is installed and the path to the 1.1.0 
+# manifest file. The script performs the following steps
+# * Backup of all the operator resources within the namespace like
+#   - service-account, deployment, role, role-binding, cluster-role, cluster-role-binding
+# * Deletes all the old Splunk operator resources and deployment
+# * Installs the operator 1.1.0 in Splunk-operator namespace. 
+# 
+# By default Splunk operator 1.1.0 will be installed to watch cluster-wide
+# Steps for upgrade from 1.0.5 to 1.1.0
+# run upgrade-to-1.1.0.sh script with below mentioned mandatory arguments 
+# current_namespace current namespace where operator is installed, if its not found, it 
+# will exit with error message
+# manifest_file: path where 1.1.0 Splunk operator manifest file exist
+#
+# example 
+# >upgrade-to-1.1.0.sh --current_namespace=splunk-operator manifest_file=release-v1.1.0/splunk-operator-cluster.yaml
+#
+#
+# Configuring Operator to watch specific namespace:
+# Edit config-map splunk-operator-config in splunk-operator namespace, set WATCH_NAMESPACE 
+# field to the namespace operator need to watch
+#
+# apiVersion: v1
+# data:
+#   OPERATOR_NAME: '"splunk-operator"'
+#   RELATED_IMAGE_SPLUNK_ENTERPRISE: splunk/splunk:latest
+#   WATCH_NAMESPACE: "add namespace here"
+# kind: ConfigMap
+# metadata:
+#   labels:
+#     name: splunk-operator
+#   name: splunk-operator-config
+#   namespace: splunk-operator
+
+readonly CURRENT_TIME=$(date +%Y-%m-%d_%H%M-%Z)
+readonly PROGRAM_NAME=$(basename "$0")
+
+help() {
+  echo ""
+  echo "USAGE: ${PROGRAM_NAME} --help [ --current_namespace=<namespacename> ] [ --manifest_file=<fulepath with filename of splunk operator 1.1.0 manfiests file>] "
+  echo ""
+  echo "OPTIONS:"
+  echo ""
+  echo -e "   --current_namespace specifiy the current namespace where operator is installed, \n" \
+       "                          script will delete existing serviceaccount, deployment, role and " \
+       "                          rolebinding and install the operator in splunk-operator namespace"
+  echo ""
+  echo "   --manifest_file splunk operator 1.1.0 manifest file path, this can be url link or full path of the file"
+  echo ""
+  echo ""
+  echo "   --help  Show this help message."
+  echo ""
+}
+
+parse_options() {
+  local count="$#"
+
+  for i in $(seq "${count}"); do
+    eval arg="\$$i"
+    param="$(echo "${arg}" | awk -F '=' '{print $1}' | sed -e 's|--||')"
+    val="$(echo "${arg}" | awk -F '=' '{print $2}')"
+
+    case "${param}" in
+      current_namespace)
+        eval "${param}"="${val}"
+        ;;
+      manifest_file)
+        eval "${param}"="${val}"
+        ;;
+      help)
+        help && exit 0
+        ;;
+      *)
+        echo "Parameter not found: '$param'"
+        help && exit 1
+        ;;
+    esac
+  done
```

Comment:
Positional parameters parsing may be an alternative clean way(?), as you don't need to parse the key-value pairs?

Reply:
issue is I need to make it work for both Mac and Linux machine, getops arguments are different in Mac/Linux

Reply:
I was referring to just the command-line options, ${1} to ${N}. So, should work for MacOS. But, it's fine, as this also works. 
BTW, do we need to have the equivalent for Windows power shell? OR, are we going to recommend using Cygwin kind of a layer?

Reply:
I think we should just go with mac/linux system. window's they can always use cygwin.